cmd/containerd-shim/main_unix.go

```diff
@@ -86,7 +91,16 @@ func main() {
 // serve serves the grpc API over a unix socket at the provided path
 // this function does not block
 func serve(server *grpc.Server, path string) error {
-	l, err := net.FileListener(os.NewFile(3, "socket"))
+	var (
+		l   net.Listener
+		err error
+	)
+	if path == "" {
```

Comment:
Well, the inheritance speeds up the connection by a good second, but when debugging just the shim it's easier to start it on the command line